commit 8a84b7248aac75a083fa3d01f99f1e6fa0e06c53
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2019-09-03T19:32:08+00:00 (5 weeks ago)
    
    TRITON-1793 Add more fabrics permission tests
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/test/integration/fabrics.test.js b/test/integration/fabrics.test.js
index bab7b8e..12b882a 100644
--- a/test/integration/fabrics.test.js
+++ b/test/integration/fabrics.test.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2018, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -863,6 +863,48 @@ test('list networks', function (t) {
 
 });
 
+test('network owner checks', function (t) {
+    t.test('get NETS[2]: correct owner (OWNERS[0])', function (t2) {
+        mod_fabric_net.get(t2, {
+            params: {
+                vlan_id: VLANS[1].vlan_id,
+                uuid: NETS[2].uuid,
+                owner_uuid: OWNERS[0]
+            },
+            exp: NETS[2]
+        });
+    });
+
+    t.test('get NETS[2]: incorrect owner (OWNERS[1])', function (t2) {
+        mod_fabric_net.get(t2, {
+            params: {
+                vlan_id: VLANS[1].vlan_id,
+                uuid: NETS[2].uuid,
+                owner_uuid: OWNERS[1]
+            },
+            expCode: 404,
+            expErr: {
+                code: 'ResourceNotFound',
+                message: 'network not found'
+            }
+        });
+    });
+
+    t.test('delete NETS[2]: incorrect owner (OWNERS[1])', function (t2) {
+        mod_fabric_net.del(t2, {
+            params: {
+                vlan_id: VLANS[1].vlan_id,
+                uuid: NETS[2].uuid,
+                owner_uuid: OWNERS[1]
+            },
+            expCode: 404,
+            expErr: {
+                code: 'ResourceNotFound',
+                message: 'network not found'
+            }
+        });
+    });
+});
 
 /*
  * Test that we can create the same network (particularly with the same name)
